### ISSUE TITLE
Loss on codebook training. What are good loss values?

### ISSUE THREAD

**Issue by gigadeplex** [TencentARC/VQFR#3]
I'm trying to train a new codebook but I'm having seemingly super low loss values recorded. The training is really slow by the way, ETA 7-8 days to reach the 800k mark. So here are my two questions:
1. Were your losses also this low when training? 
2. What made you decide on the 800k?
 
`[train..][epoch:  0, iter:  28,500, lr:(1.000e-04,1.000e-04,)] [eta: 7 days, 21:16:33, time (data): 0.782 (0.003)] l_rec: 7.4355e-03 l_g_percep: 3.9922e-02 l_codebook: 5.1051e-03 l_total_g: 5.2463e-02 
`

**Comment by guyuchao** [comment]
1) The loss seems to have some problems. Have you changed any configurations and do you run the code in batch=16? 
`[epoch:  6, iter:  28,500, lr:(1.000e-04,)] [eta: 3 days, 12:00:50, time (data): 0.330 (0.003)] l_rec: 9.6236e-02 l_g_percep: 3.1971e-01 l_codebook: 6.3252e-02 l_total_g: 4.7919e-01`
2) We decide the 800K following the configuration of GFP-GAN and Restoreformer. Since the reconstruction results will be stable after about 500K, maybe you can try smaller epochs. In 8 x A100 machines, the total codebook training cost is about 3 day, 13 hours. If you use V100 machines, it may requires about twice training time.

**Comment by gigadeplex** [comment]
I changed the dataset from FFHQ to full images instead of just faces and I changed the color space to LAB. I also changed the configuration as well as I have 4 GPUs not 8 and their max memory is 24,5 GB each. So the batch size per gpu is 3. Is it a must that it is in RGB color space? and can it manage full images instead of just faces?

**Comment by guyuchao** [comment]
It seems you try to adapt this codebase to other tasks. Because we have not try other tasks, we cannot give informative instructions. But I can provide several suggestions for you: 1) the lpips perceptual loss receives input of RGB color space. 2) The batch size need to be large for real images, otherwise, it requires a smaller learning rate and enlarges training epochs (refer to VQGAN).

**Comment by gigadeplex** [comment]
1) Thank you so much, I can't believe I missed that.
2) To increase the batch size I must decrease the codebook size, how much will this impact the quality do you think?

**Comment by guyuchao** [comment]
For 2): It depends on the task you would like to do. Sorry for not providing more useful information. If there are any other questions, feel free to reopen it.
